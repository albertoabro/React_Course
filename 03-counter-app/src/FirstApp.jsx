
import PropTypes from 'prop-types';

/*For const values or not hooks relationship, write the instruction out.
this sentence it's encapsulated in this module
*/
const newMessage = 'Alberto' //Boolean values, undefined and null not will be printed / Object are not valid 
const object = { //Objects and Promises must be sending with JSON.stringify(nameObject)
    message: 'message',
    message2: 'message2'
}

export const App = ({title: title = 'Not title', subTitle = 'Not subtitle', name = 'Alberto'}) => {

    
    return (
        // <> it's equal to Fragment -> Group text's elements, map collection to Fragment's array, assign multi element to once variable 
        <> 
        <h1 data-testid="test-title">{title}</h1> {/* For use Js code, use {} and for comments inside of JSX use like this*/}
        <code>{JSON.stringify(object)}</code>
        <p>{subTitle}</p>
        <p>{subTitle}</p>
        <p>{name}</p>
        </>
    )
}

App.propTypes = {
    title: PropTypes.string.isRequired,
}