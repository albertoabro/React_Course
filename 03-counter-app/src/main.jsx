import React, { StrictMode } from 'react'
import { createRoot } from 'react-dom/client'
//import { App } from './FirstApp'
import { CounterApp } from './CounterApp'
import './index.css' //For global Styles

createRoot(document.getElementById('root')).render(
    <StrictMode>
        <CounterApp value={100}/>
        {/* <App title='Hello world'/> */}
    </StrictMode>
)