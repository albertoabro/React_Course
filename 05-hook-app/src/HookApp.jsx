import '@fontsource-variable/montserrat';

export const HookApp = () => {
    return (
        <h1>Hook App</h1>
    )
} 