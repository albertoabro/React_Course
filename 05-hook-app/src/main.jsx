import { StrictMode } from 'react'
import { createRoot } from 'react-dom/client'
import { BrowserRouter } from 'react-router';
import '@fontsource-variable/montserrat';
import './index.css'
import { MainApp } from './09-useContext/MainApp';

//import { TodoApp } from './08-useReducer/TodoApp';
//import { Padre } from './07-tarea-memo/Padre';
//import './08-useReducer/intro-reducer'
//import { CallBackHook } from './06-memos/CallBackHook';
//import { MemoHook } from './06-memos/MemoHook';
//import { Memorize } from './06-memos/Memorize';
//import { FocusScreen } from './04-useRef/FocusScreen';
//import { MultipleCustomHook } from './03-examples/MultipleCustomHook'
//import { FormWithCustomHook } from './02-useEffect/FormWithCustomHook'
//import { SimpleForm } from './02-useEffect/SimpleForm'
//import { CounterWithCustomHook } from './01-useState/CounterWithCustomHook'
//import {CounterApp} from './01-useState/CounterApp'
//import {HookApp} from '@/HookApp.jsx'


createRoot(document.getElementById('root')).render(
  <BrowserRouter>
  <StrictMode>
    <MainApp />
  </StrictMode>
  </BrowserRouter>
)
