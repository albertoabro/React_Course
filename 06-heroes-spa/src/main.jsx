import { StrictMode } from 'react'
import { createRoot } from 'react-dom/client'
import { BrowserRouter } from 'react-router'
import './styles.css'
import { HeroesApp } from './HeroesApp.jsx'

createRoot(document.getElementById('root')).render(
  <BrowserRouter>
    <StrictMode>
      <HeroesApp />
    </StrictMode>
  </BrowserRouter>
)
