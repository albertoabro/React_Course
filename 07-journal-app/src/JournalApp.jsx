import { AuthProvider } from "./auth/context/AuthProvider"
import { AppRouter } from "./router/AppRouter"
import { AppTheme } from "./theme/AppTheme"


export const JournalApp = () => {
  return (

    <AppTheme>
    <AppRouter />
    </AppTheme>

  )
}