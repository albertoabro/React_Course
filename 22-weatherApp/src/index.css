:root {
    --white: #fff;
    --primary: #f59e0b;
}

html {
    font-size: 62.5%;
}

body {
    background-image: url('/public/bg_clima.jpg');
    background-repeat: no-repeat;
    background-size: cover;
    color: var(--white);
    font-family: "Outfit", sans-serif;
    font-size: 2rem;
    height: 100vh;
}
