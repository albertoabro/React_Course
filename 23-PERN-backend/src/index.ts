import * as colors from 'colors';
import 'dotenv/config';
import server from './infrastructure/adapter/server';

server.listen(process.env.SERVER_PORT, () => {
    console.log( colors.cyan.bold(`REST Api Server is running on port ${process.env.SERVER_PORT}`));
})