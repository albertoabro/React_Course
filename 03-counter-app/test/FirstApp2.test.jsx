import { render, screen } from "@testing-library/react"
import { App } from "../src/FirstApp"


describe('Test in <App/>', () => {

    const title = 'Hello world';
    const subTitle = 'I am subtitle';

    test('should do match with snapshot', () => {
        
        const {container} = render( <App title={title}/>);

        expect(container).toMatchSnapshot();
    });

    test('should show the message "Hello World"', () => {
        render( <App title={title}/>);
        expect(screen.getByText(title)).toBeTruthy();
    });

    test('should show the title in <h1>', () => {
        render( <App title={title}/>);
        //GetByRol obtain a HTML elements
        expect(screen.getByRole('heading', {level: 1}).innerHTML).toContain(title);
    });
    
    test('should show the subtitle was sent by props ', () => {
        
        render( 
            <App 
                title={title}
                subTitle={subTitle}
            />
        );

        expect( screen.getAllByText(subTitle).length).toBe(2);
    });
})
