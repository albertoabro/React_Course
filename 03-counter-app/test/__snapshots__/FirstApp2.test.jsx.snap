// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <App/> should do match with snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hello world
  </h1>
   
  <code>
    {"message":"message","message2":"message2"}
  </code>
  <p>
    Not subtitle
  </p>
  <p>
    Not subtitle
  </p>
  <p>
    Alberto
  </p>
</div>
`;
